test/ttmlir/Dialect/TTNN/multiple_add_with_loc_grid_override.mlir

```diff
@@ -0,0 +1,28 @@
+// RUN: ttmlir-opt --ttir-to-ttnn-backend-pipeline="override-grid-sizes=add_1_0=4x4,add_2_0=4x4" %s | FileCheck %s
+#any_device = #tt.operand_constraint<dram|l1|scalar|tile|any_device|any_device_tile>
+#loc = loc("test_ops.py:17_0_0":0:0)
```

Comment:
Few questions around locations in the test:
- does this string represent callsite? 
- do we need locations for each tensor?
- those `loc(#other_loc)` are confusing as well, what's the reason for chaining them?

Reply:
Test input MLIR below is as generated by Pybuda for small demo we had on company team meeting. Per spec they are mandatory to be generated when lowering to MLIR, if not defined loc(unknown) will be set.
Yes they represent callsite.
Each Frontend defines during lowering to MLIR how location data should be populated, so we can define whatever we think is needed here and strive to have all Frontends in sync.
Chaining is per MLIR Location builtin dialect. There are several types of location data, like FileLineColLoc, NameLoc, FusedLoc... NameLoc is created by passing in childloc and assigning a name which is good for naming Nodes/Operations like we did in Buda.
More information within MLIR builtin dialect spec: https://mlir.llvm.org/docs/Dialects/Builtin/#location-attributes